src/Analysis/GWASV2/Shared/GWASCovariates/CustomDichotomousCovariates.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import CohortSelect from '../../SelectStudyPopulation/Utils/CohortSelect';
+import '../../../GWASUIApp/GWASUIApp.css';
+
+const CustomDichotomousCovariates = ({
+  handleCovariateSubmit,
+  sourceId,
+  current,
+  setMode,
+}) => {
+  const [firstPopulation, setFirstPopulation] = useState(undefined);
+  const [secondPopulation, setSecondPopulation] = useState(undefined);
+  const [providedName, setProvidedName] = useState('');
+
+  const handleDichotomousSubmit = () => {
+    handleCovariateSubmit({
+      uuid: _.uniqueId(),
+      variable_type: 'custom_dichotomous',
+      cohort_ids: [firstPopulation.cohort_definition_id, secondPopulation.cohort_definition_id],
+      provided_name: providedName,
+    });
+    setMode('');
+  };
+
+  const customDichotomousValidation = providedName.length === 0 || firstPopulation === undefined || secondPopulation === undefined;
+
+  return (
+    <div>
+      <div className='GWASUI-flexRow' data-tour='name'>
+        <input
+          type='text'
+          className={'GWASUI-providedName'}
+          onChange={(e) => setProvidedName(e.target.value)}
+          value={providedName}
+          placeholder='Provide a name...'
+          style={{
+            width: '50%', textSize: 'small', paddingLeft: 5, height: 45, borderRadius: 5, marginTop: 5,
+          }}
+        />
+        <button type='button' className={'GWASUI-dichBtn'} onClick={() => setMode('')}>
+          cancel
+        </button>
+        <div data-tour='add-button'>
+          <button
+            type='button'
+            disabled={customDichotomousValidation}
+            className={`${!customDichotomousValidation ? 'GWASUI-btnEnable' : ''} GWASUI-dichBtn`}
```

Comment:
I think this can be simplified to 
```
            className={`${!customDichotomousValidation && 'GWASUI-btnEnable'} GWASUI-dichBtn`}
```
(try it first, I'm not 100% sure)

Reply:
yeah i agree, i think we should make an agnostic btn component that takes a `boolean ` prop